docs/standard/library-guidance/breaking-changes.md

```diff
@@ -0,0 +1,98 @@
+
+  Building block packages are used by end-user developers to build applications, and by other projects as NuGet dependencies. For example, you're building an application and are using an open-source client to call a web service. A breaking update to dependency the client uses isn't something you can fix. It's the open-source client that needs to be changed and you have no control over it. You have to find compatible versions of the libraries, or submit a fix to the client library and wait for a new version. The worst-case situation is if you want to use two libraries that depend on mutually incompatible versions of a third library.
+
+* **High-level libraries** like a suite of UI controls are less sensitive to breaking changes.
+
+  High-level libraries are directly referenced in an end-user application. If breaking changes occur the developer can choose to update to the latest version, or can modify their application to work with the breaking change.
+
+**✔️ DO** think about how your library will be used. What effect will breaking changes have on applications and libraries that use it?
+
+**✔️ DO** minimize breaking changes when developing a low-level .NET library.
+
+**✔️ CONSIDER** publishing a major rewrite of a library as a new NuGet package.
+
+## Types of breaking changes
+
+Breaking changes fall into different categories and aren't equally impactful.
+
+### Source breaking change
+
+A source breaking change doesn't affect program execution but will cause compilation errors the next time the application is recompiled. For example, a new overload can create an ambiguity in method calls that were unambiguous previously, or a renamed parameter will break callers that use named parameters.
+
+```csharp
+public class Task
+{
+    // Adding a type called Task could conflict with System.Threading.Tasks.Task at compilation
+}
+```
+
+Because a source breaking change is only harmful when the developer recompiles their application, it's the least disruptive breaking change. Developers can fix their own broken source code easily.
+
+### Behavior breaking change
+
+Behavior changes are the most common breaking change: almost any change in behavior could break someone. Changing the results a method produces, the exception your library throws, or the data output it writes to the network or disk could all negatively impact applications and libraries that depend on you. Even a bug fix can qualify as a breaking change if users relied on the previously broken behavior.
+
+Adding features and improving bad behaviors is a good thing, but without care it can make it very hard for existing software to upgrade. One approach to helping developers deal with behavior breaking changes is to hide them behind settings. Settings let developers update to the latest version of your library while at the same time choosing to opt in or opt out of breaking changes. This strategy lets developers stay up-to-date while letting their consuming code adapt over time.
```

Comment:
> ...very hard for existing software to upgrade. 

There seems to be a constant change of used terms, "libraries", "applications", "software", "assemblies". This creates a cognitive overload, which is quite undesirable for a documentation.
Perhaps it is best to use one term consistently across the document.
In this case, coupled with the above suggestion, perhaps reword it to "_....very hard for existing consumers to upgrade._"?